generators/entity-client/templates/angular/src/main/webapp/app/entities/_entity.model.ts

```diff
@@ -70,11 +70,32 @@ relationships.forEach(relationship => {
             }
             fieldName = relationship.relationshipFieldName;
         } else {
-            fieldType = tsKeyType;
-            fieldName = `${relationship.relationshipFieldName}Id`;
+            const otherEntityRelationshipName = relationship.otherEntityRelationshipName;
```

Comment:
@gmarziou what we could do to avoid duplication is to move this logic to a js function in our generator-private-base and then just call it here. The function will take `relationships` and `fields` as argument and just return the required variables as an object map. WDYT

Reply:
This would be great, I can try for v5.